src/Tools/ExternalAccess/LiveShare/CodeActions/RoslynCodeActionProvider.Exports.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.LiveShare.CodeActions
+{
+    [Shared]
+    [ExportCodeRefactoringProvider(StringConstants.CSharpLspLanguageName)]
+    internal class CSharpLspCodeActionProvider : RoslynCodeActionProvider
+    {
+        [ImportingConstructor]
```

Comment:
probably typescript never use this?

Reply:
yep typescript liveshare client is not using this in 16.0